### PR TITLE
boards/common/{nrf52,cc26xx_cc13xx}: improve default programmer selection

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

For nRF52xxx and CC26xx and CC13xx MCUs J-Link and uniflash are currently chosen as default flashers, without checking if they are available. This adds a check and only picks them as default if they are available, falling back to OpenOCD when not. Behavior should remain unchanged for happy users of J-Link and uniflash, but life should be better for those who prefer to stick with OpenOCD.

### Testing procedure

`make flash` should now work out of the box when OpenOCD is installed but J-Link and uniflash are not, but should not change when there are available.

### Issues/PRs references

None

### CODE REVIEW

File: boards/common/nrf52/Makefile.include [hunk 1]
```diff
@@ -8,7 +8,11 @@ ifeq (bmp,$(PROGRAMMER))
   PORT_DARWIN ?= $(wordlist 2, 2, $(sort $(wildcard /dev/tty.usbmodem*)))
 endif
 
-PROGRAMMER ?= jlink
+ifneq (,$(command -v jlink))
+  PROGRAMMER ?= jlink
```
Comment (aabadie) [inline boards/common/nrf52/Makefile.include:12]: I don't think this works. The jlink command is not jlink (at least on Linux).

https://github.com/RIOT-OS/RIOT/blob/c485bd4d92a1acf81195eb47961317d58b8319bd/dist/tools/jlink/jlink.sh#L71

Comment (maribu) [reply boards/common/nrf52/Makefile.include:12]: Good catch!
```suggestion
  PROGRAMMER ?= JLinkExe
```

Comment (jeandudey) [reply boards/common/nrf52/Makefile.include]: Shouldn't it be backwards?  e.g.: `ifneq (,$(command -v JLinkExe))` and `PROGRAMMER ?= jlink`.

I would also add an additional `ifneq (,$(command -v $JLINK))` if the user specifies the `JLinkExe` executable manually without having to set `PROGRAMMER`. e.g.: `JLINK=/path/JLinkExe make flash`

Comment (maribu) [reply boards/common/nrf52/Makefile.include:14]: > Shouldn't it be backwards? e.g.: `ifneq (,$(command -v JLinkExe))` and `PROGRAMMER ?= jlink`.

Absolutely :) I confused the lines :/

> I would also add an additional `ifneq (,$(command -v $JLINK))` if the user specifies the `JLinkExe` executable manually without having to set `PROGRAMMER`. e.g.: `JLINK=/path/JLinkExe make flash`

Done.

File: boards/common/nrf52/Makefile.include [hunk 2]
```diff
@@ -8,7 +8,13 @@ ifeq (bmp,$(PROGRAMMER))
   PORT_DARWIN ?= $(wordlist 2, 2, $(sort $(wildcard /dev/tty.usbmodem*)))
 endif
 
-PROGRAMMER ?= jlink
+% keep name of `JLINK` in sync with script jlink.sh in $(RIOTTOOLS)/jlink
```
Comment (jeandudey) [inline boards/common/nrf52/Makefile.include:11]:
```suggestion
# keep name of `JLINK` in sync with script jlink.sh in $(RIOTTOOLS)/jlink
```

Comment (maribu) [reply boards/common/nrf52/Makefile.include]: Looks like I have been using too much LaTeX lately (:

Fixed, thanks!

**Review by jeandudey — APPROVED** [review]
LGTM

**Comment by maribu** [comment]
Thanks for the review! :)

**Pull request merged**